### PR TITLE
fix(auth): catch refresh-token failures in defineApiHandler

### PR DESCRIPTION
**Pull request by baktun14**
## Why

When Auth0's `getSession` throws (e.g. `ERR_FAILED_REFRESH_GRANT`, `ERR_EXPIRED_ACCESS_TOKEN`, corrupt session cookie), the `defineApiHandler` chain crashes before any route runs. The `/api/proxy/[...path]` route is wrapped by `defineApiHandler` and its own `proxyRequest` catch path converts thrown errors into `502`s — so users whose refresh token expired (or whose session cookie is invalid) see `502` when they try to create a deployment instead of the clean `401` they need to be redirected to login.

Refs #2465 — and is the same family of failures originally seen in #2148 (`ProfileHandlerError ... refresh token is not available`).

## What

- `defineApiHandler` now wraps `getSession` in `try/catch`:
  - On `AccessTokenError` with code `EXPIRED_ACCESS_TOKEN` or `FAILED_REFRESH_GRANT`: log `API_SESSION_INVALID`, clear `appSession*` cookies, pass `session = null` to the route handler.
  - On any other unexpected error: report via `errorHandler` with `event: API_GET_SESSION_ERROR`, pass `session = null`.
- Routes (proxy, profile, etc.) get a usable context. The proxy now reads `session` from context (was redundantly calling `getSession` again) and forwards the request without the `Authorization` header when the session is gone — the upstream API returns a clean `401` which the proxy faithfully proxies. `502` is reserved for genuine upstream failure, as the existing `proxyRequest.spec.ts` contract intends.
- Extract `clearSessionCookies` and `isInvalidSessionError` into shared utils (`src/lib/auth0/...`), replacing the inline duplicates in `[...auth0].ts`. The local-Auth0 logout path also uses the shared helper now.
- `proxyRequest` is registered on the server DI container so `[...path]` can be unit-tested via the existing `REQ_SERVICES_KEY` injection pattern.
- `vitest.config.ts`: route `src/pages/api/**/*.spec.ts` to the `unit-node` project so the new `[...path].spec.ts` runs in the right environment.

## Test plan
- [x] `npx vitest run src/lib/auth0/clearSessionCookies/...` — 3 new tests
- [x] `npx vitest run src/lib/auth0/isInvalidSessionError/...` — 5 new tests
- [x] `npx vitest run src/lib/nextjs/defineApiHandler/...` — 3 new tests cover `EXPIRED_ACCESS_TOKEN`, `FAILED_REFRESH_GRANT`, and unexpected errors
- [x] `npx vitest run "src/pages/api/proxy/[...path].spec.ts"` — 4 new tests cover Bearer forwarding behavior
- [x] `npm run test:unit` — only pre-existing failures remain (unrelated `TrendIndicator` test, fails on `main` too)
- [x] `npx tsc --noEmit` — no new TS errors in touched files
- [x] `npx eslint --quiet` on touched files — clean
- [ ] Manual smoke: corrupt the `appSession` cookie in DevTools → reload → trigger a deployment action → confirm browser receives **401** (not 502), the bad cookie is cleared via `Set-Cookie`, and the app routes to login

## Summary by CodeRabbit

* **New Features**
  * Added utilities to detect invalid sessions and to clear stale session cookies; proxy handler now accepts session info and DI container exposes proxy functionality.

* **Bug Fixes**
  * Improved handling of expired/invalid authentication sessions with automatic cookie cleanup, warning logs for invalid sessions, and safer session retrieval with error reporting.

* **Tests**
  * Added unit tests covering session invalidation, cookie clearing, API proxy behavior, header forwarding, and edge cases.

* **Chores**
  * Updated test configuration to include API tests and adjust coverage exclusions.

### CODE REVIEW

File: apps/deploy-web/src/lib/auth0/clearSessionCookies/clearSessionCookies.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export function clearSessionCookies(req: NextApiRequest, res: NextApiResponse): void {
+  const cookies = req.cookies;
+  const expiredCookies = Object.keys(cookies)
+    .filter(key => key.startsWith(SESSION_COOKIE_PREFIX))
+    .map(key => `${key}=; ${EXPIRED_COOKIE_OPTIONS}`);
+
+  if (expiredCookies.length > 0) {
+    res.setHeader("Set-Cookie", expiredCookies);
+  }
```
[the remark this answers is not in the document]
Comment (baktun14) [reply apps/deploy-web/src/lib/auth0/clearSessionCookies/clearSessionCookies.ts:16]: Fixed in 7e34cc590c6d3fa0c9b1c41233c49f84e95693cd — clearSessionCookies now reads any existing Set-Cookie header(s), normalizes to an array, and appends the expired cookies.

File: apps/deploy-web/src/pages/api/proxy/[...path].spec.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    const { proxyRequest } = await setup({ session: null });
+
+    expect(getForwardedHeaders(proxyRequest).authorization).toBeUndefined();
+  });
+
+  it("forwards cf-connecting-ip and request-id headers", async () => {
+    const { proxyRequest } = await setup({ session: null });
+
+    expect(getForwardedHeaders(proxyRequest)["cf-connecting-ip"]).toBe("127.0.0.1");
+  });
```
[the remark this answers is not in the document]
Comment (baktun14) [reply apps/deploy-web/src/pages/api/proxy/[...path].spec.ts]: The proxy handler does not forward a request-id header (only cf-connecting-ip, traceparent, sentry-trace, baggage, cookie, authorization). Renamed the test in 7e34cc590c6d3fa0c9b1c41233c49f84e95693cd to drop "and request-id headers" so it matches what is actually asserted; adding request-id forwarding is out of scope for this PR.

File: apps/deploy-web/src/pages/api/proxy/[...path].spec.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      url: "/api/proxy/v1/tx",
+      method: "POST",
+      headers: {} as NextApiRequest["headers"],
+      cookies: {},
+      socket: { remoteAddress: "127.0.0.1" } as Socket
+    } as unknown as NextApiRequest;
+    const res = mock<NextApiResponse>();
+
+    (req as NextApiRequestWithServices)[REQ_SERVICES_KEY] = {
+      ...services,
```
[the remark this answers is not in the document]
Comment (baktun14) [reply apps/deploy-web/src/pages/api/proxy/[...path].spec.ts]: Fixed in be34444e408806359dae8e631c0606c0e045d979 — req is now built with mock<NextApiRequestWithServices>(). Note: vitest-mock-extended recursively wraps override values in proxies that auto-mock any property access, so headers can't be passed as an override (unknown lookups like req.headers["sentry-trace"] would return mock fns instead of undefined). Reassigning req.headers post-construction stores the plain object directly via the proxy's set trap and bypasses that.

**Review by stalniy — APPROVED** [review]
